images/macos/provision/utils/utils.sh

```diff
@@ -173,6 +173,38 @@ brew_smart_install() {
         brew install --build-from-source $tool_name
     else
         echo "Downloading $tool_name..."
+
+        # get deps & cache em
+
+        failed=true
+        for i in {1..10}; do
+            brew deps $tool_name > /tmp/$tool_name && failed=false || sleep 60
+            if [ "$failed" = false ]; then
+                break
+            fi
+        done
+
+        if [ "$failed" = true ]; then
+           echo "Failed: brew deps $tool_name"
+           exit 1;
+        fi
```

Comment:
It's possible to implement this logic simpler:

- place `break` in place of `failed=false`;
- change condition in the inner 'if' block to `$i -eq 10` and put `exit` inside;
- remove `failed` variable and outer 'if' block

Something like this:
```
for i in {1..10}; do
    brew deps $tool_name > /tmp/$tool_name && break || sleep 60
    if [ $i -eq 10 ]; then
        echo "Failed: brew deps $tool_name"
        exit 1
    fi
done
```

Reply:
I tried inline "break", it did not work on my syntetic tests.
if you end with something working, feel free to submit follow up PR

Reply:
personally, I feel that `if [ $i -eq 10 ]; then` is not "readable", it may be harder to understand it's intention